### ISSUE TITLE
[Core] Update dependency version of System.Memory.Data to consume new MediaType property

### ISSUE THREAD

**Issue by JoshLove-msft** [Azure/azure-sdk-for-net#41941]
The MediaType property is available 8.0.0. We need it for codegen https://github.com/Azure/autorest.csharp/issues/3731 as well as for other use cases potentially in System.ClientModel.

**Comment by m-redding** [comment]
Having an old reference creates some AOT warnings since the old package is not annotated - https://github.com/Azure/azure-sdk-for-net/issues/44127

**Comment by m-redding** [comment]
There is a bug in version 6.0.0 (https://github.com/Azure/azure-sdk-for-net/pull/46134)

Remove workarounds added to Azure.Storage.Queues in https://github.com/Azure/azure-sdk-for-net/pull/46134  when this package is upgraded to 8.0.0.

**Comment by annelo-msft** [comment]
@m-redding, is this issue unblocked now with the SMD upgrade?  Or is there still work needed to unblock?

**Comment by m-redding** [comment]
@annelo-msft No we're stuck on 6.0.0 because of Functions/Powershell. We may be able to move to 8.0.0 in November depending on Functions.

**Comment by cataggar** [comment]
@m-redding, any update on bumping to System.Memory.Data? Version 6.0.0 bit us pretty bad #47544.

**Comment by mirekkukla** [comment]
Would love to see this fixed. Also arrived after getting bit by #47544

**Comment by jsquire** [comment]
Closing as a duplicate of #46280
